string_manipulation.rb

```diff
@@ -1,30 +1,127 @@
 # A method to reverse a string in place.
 def string_reverse(my_string)
-  puts "NOT IMPLEMENTED"
+  length = my_string.length
+  i = 0
+
+  if length >= 2
+  (length / 2).times do
+    temp = my_string[i]
+    my_string[i] = my_string[length - 1]
+    my_string[length - 1] = temp
+    i += 1
+    length -= 1
+    end
+  end
+  return my_string
 end
 
 # A method to reverse each word in a sentence, in place.
 def reverse_words(my_words)
-  puts "NOT IMPLEMENTED"
+  length = my_words.length
+
+  word_start = 0
+  word_end = 0
+
+  length.times do |i|
+    if my_words[i] == " " || i == length - 1
+
+      if my_words[i] == " "
+        word_end = i - 1
+      else
+        word_end = i
+      end
+
+      my_words[word_start..word_end] = string_reverse(my_words[word_start..word_end])

```

Comment:
my_words[word_start..word_end] creates a new string object and then that new object gets reversed in place by string_reverse. You are then assigning that object back to my_words[word_start..word_end]. Creating a new string object will add to your space complexity. See if you can avoid this.